mcc/odin/odin/snapshot_list_test.go

```diff
@@ -17,3 +21,39 @@ var exampleSnapshot1 = &rds.DBSnapshot{
 	MasterUsername:       aws.String("owner"),
 	Status:               aws.String("available"),
 }
+
+type listSnapshotsCase struct {
+	testCase
+	name       string
+	snapshots  []*rds.DBSnapshot
+	instanceID string
+}
+
+var listSnapshotsCases = []listSnapshotsCase{
+	// No snapshots for any instance
+	{
+		testCase: testCase{
+			expected:      []*rds.DBSnapshot{},
+			expectedError: "",
+		},
+		name:       "No snapshots for any instance",
+		snapshots:  []*rds.DBSnapshot{},
+		instanceID: "",
+	},
+}
+
+func TestListSnapshots(t *testing.T) {
+	svc := newMockRDSClient()
+	for _, test := range listSnapshotsCases {
+		t.Run(
+			test.name,
+			func(t *testing.T) {
+				actual, err := odin.ListSnapshots(
+					test.instanceID,
+					svc,
+				)
+				test.check(actual, err, t)
```

Comment:
`check` is a method from `testCase` that helps to check for unexpected errors happening, or expected errors not happening, or comparing the `actual` results with the `expected` results, which are defined in the test case.
`t` is just the `testing` interface to report failures when happening.